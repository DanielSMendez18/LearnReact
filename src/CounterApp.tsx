import { useState } from "react"

interface Item {
    name: string,
    value: number
}

const CounterApp = ({name, value} : Item) => {

    // count es la variable de estado actual, que comienza con un valor de 0.
    // setCount es la función para actualizar el estado de count.
    const [counter, setCounter] = useState(value);

    // Al hacer clic en el botón, llamamos a setCount(count + 1) para incrementar el valor de count.
    const handleAdd = () => {
        //console.log(event)
        setCounter( counter + 1)
    }

    const handleRemove = () => {
       if (counter > 0) {
        setCounter (counter - 1)
       }
    }

    const handleReset = () => {
        setCounter(value)
    }

    return(
        <>
            <h1>Counter App</h1>
            <h3> Item: {name}</h3>
            <h3> Value: {counter}</h3>
            <button onClick={handleAdd}> Add </button> 
            <button onClick={handleRemove}> Remove </button> 
            <button onClick={handleReset}> Reset </button> 
        </>
    )
}

export default CounterApp;