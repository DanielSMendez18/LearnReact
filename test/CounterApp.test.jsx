import React from 'react'
import { render } from '@testing-library/react'
import CounterApp from "../src/CounterApp";

describe('@CounterApp', () => {
  test('#should increment the counter by 1', () => {
    const {container} = render(<CounterApp name="nombredemo" value={4}/>);
    expect(container).toMatchSnapshot(); 
  })
})
