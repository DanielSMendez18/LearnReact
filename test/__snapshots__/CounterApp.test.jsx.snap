// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@CounterApp #should increment the counter by 1 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h3
    id="itemText"
  >
     Item: 
    nombredemo
  </h3>
  <h3
    data-testid="test-value"
  >
     New Value: 
    4
  </h3>
  <h4>
    nombredemo
  </h4>
  <button>
     Add 
  </button>
  <button>
     Remove 
  </button>
  <button>
     Reset 
  </button>
</div>
`;

exports[`@CounterApp 2 #should to make match with the snapshop 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h3
    id="itemText"
  >
     Item: 
    Misión rescate
  </h3>
  <h3
    data-testid="test-value"
  >
     New Value: 
    1
  </h3>
  <h4>
    Misión rescate
  </h4>
  <button>
     Add 
  </button>
  <button>
     Remove 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
